concordium-contracts-common-derive/src/lib.rs

```diff
@@ -44,33 +44,50 @@ pub fn deserial_derive(input: TokenStream) -> TokenStream {
     unwrap_or_report(derive::impl_deserial(&ast))
 }
 
-/// Derive the Serial trait for the type.
+/// Derive the [`Serial`] trait for the type.
 ///
-/// If the type is a struct all fields must implement the Serial trait. If the
-/// type is an enum then all fields of each of the enums must implement the
-/// Serial trait.
+/// If the type is a struct all fields must implement the [`Serial`] trait. If
+/// the type is an enum then all fields of each of the enums must implement the
+/// [`Serial`] trait.
 ///
+/// Fields of structs are serialized in the order they appear in the code.
+///
+/// Enums can have no more than 65536 variants. They are serialized by using a
+/// tag to indicate the variant, enumerating them in the order they are written
+/// in source code. If the number of variants is less than or equal 256 then a
+/// single byte is used to encode it. Otherwise two bytes are used for the tag,
+/// encoded in little endian.
+///
+/// ## Generic type bounds
+///
+/// By default a trait bound is added on each generic type for implementing
+/// [`Serial`]. However, if this is not desirable, the bound can be put
+/// explicitly using the `bound` attribute on the type overriding the default
+/// behavior.
+///
+/// ### Example
+/// ```ignore
+/// #[derive(Serial)]
+/// #[concordium(bound(serial = "A: SomeOtherTrait"))]
+/// struct Foo<A> {
+///     bar: A,
+/// }
+/// ```
+///
+/// ## Collections
 ///
 /// Collections (Vec, BTreeMap, BTreeSet) and strings (String, str) are by
 /// default serialized by prepending the number of elements as 4 bytes
 /// little-endian. If this is too much or too little, fields of the above types
 /// can be annotated with `size_length`.
 ///
 /// The value of this field is the number of bytes that will be used for
-/// encoding the number of elements. Supported values are 1, 2, 4, 8.
+/// encoding the number of elements. Supported values are `1,` `2,` `4,` `8`.
```

Comment:
Why not?
```suggestion
/// encoding the number of elements. Supported values are `1, 2, 4, 8`.
```

Reply:
My intention was to emphasize that these are not string literals. The comma should not have been part of the monospacing

Reply:
Ok